### PR TITLE
Service portal :  Rewrite document client

### PR DESCRIPTION
**Pull request by obmagnusson**
# \<Description\>
Rewrite document REST client to user HttpModule instead of openapi generated code from an .yml file.
Rehydration of token implemented.

## What

REST client credentials communication with Island.is Postholf.

## Why

Previous setup did not allow for rehydration of tokens. And this should be a much more efficient.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

File: libs/api/domains/documents/src/lib/client/document.connection.ts
Comment (atligudl) [inline libs/api/domains/documents/src/lib/client/document.connection.ts:26]: Remove debug logs?

File: libs/api/domains/documents/src/lib/client/documentClient.ts
Comment (atligudl) [inline libs/api/domains/documents/src/lib/client/documentClient.ts:40]: const?

File: libs/api/domains/documents/src/lib/client/documentClient.ts
Comment (atligudl) [inline libs/api/domains/documents/src/lib/client/documentClient.ts:53]: console log

**Review by atligudl — COMMENTED** [review]
Few minor comments

File: libs/api/domains/documents/src/lib/client/document.connection.ts
Comment (obmagnusson) [reply libs/api/domains/documents/src/lib/client/document.connection.ts:26]: no as i understand these only fire on debugging.

**Comment by eirikurn** [comment]
Fyi, looks to me like we can implement token rehydration in a client middleware (as part of the codegen api configuration).

The "pre" hook receives the "url" and "init" fetch arguments, and can change do async logic to change any part of the request. Your call.

**Review by obmagnusson — COMMENTED** [review]
I tried that at one point. Ran into some problems at with the client not properly picking up the middleware. Guessing the yml structure was improperly set up. Maybe something we will revisit when the Postholf service gets reworked and integrated into X-Road.

**Review by andriandresson — APPROVED** [review]
LGTM 🎉

**Review by atligudl — APPROVED** [review]

**Review by arnifreyrs — APPROVED** [review]
From my codeowner standpoint this looks good.

**Pull request merged**
